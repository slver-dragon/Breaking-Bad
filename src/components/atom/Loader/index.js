import React from "react";
import style from "./style.module.scss";

const Loader = () => {
  return (
    <div className={style.container}>
      <div className={style.loading} />
    </div>
  );
};

export default Loader;
