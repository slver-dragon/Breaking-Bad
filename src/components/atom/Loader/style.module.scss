@import "../../../assets/styles/variables.scss";

.fetching {
  width: 7rem;
  height: 7rem;
  border-radius: 100%;
  border: 0.125rem dashed $red;
  position: absolute;
  top: 30%;
  left: 50%;
  animation-name: search-fething;
  animation-duration: 3s;
  animation-iteration-count: infinite;
  animation-timing-function: linear;
}
@keyframes search-fething {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
