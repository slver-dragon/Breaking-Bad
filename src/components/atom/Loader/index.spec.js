import React from "react";
import { shallow } from "enzyme";
import Loader from ".";

describe("Loader atom testing:", () => {
  test("Is render", () => {
    const component = shallow(<Loader />);
    expect(component).toMatchSnapshot();
  });
});
