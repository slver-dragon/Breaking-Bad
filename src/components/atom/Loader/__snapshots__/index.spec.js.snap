// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader atom testing: Is render 1`] = `
<div
  className="container"
>
  <div
    className="loading"
  />
</div>
`;
